internal/impl/flake.go

```diff
@@ -0,0 +1,78 @@
+package impl
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+func (d *Devbox) installNixProfileFlakeCommand(profileDir string) *exec.Cmd {
+	cmd := exec.Command(
+		"nix", "profile", "install",
+		"--profile", profileDir,
+		"--extra-experimental-features", "nix-command flakes",
+		filepath.Join(d.configDir, ".devbox/gen/flake/"), // installables
+	)
+	if d.hasDevboxLockFile() {
+		_ = d.copyDevboxLockToFlakeLock()
+		cmd.Args = append(cmd.Args, "--no-write-lock-file")
```

Comment:
Help me understand this: if the user already has a lock file, we don't want the `nix profile install` command to create a lock file?

Reply:
the second one!

Reply:
But what if the packages field has changed? Does `nix profile` overwrites the old?

Reply:
when one does `devbox add`, then the `nix profile --install` is run.

btw, right now there's some issue with this. I need to iterate on this PR (hence I was anxious to get it in).
```
❯ DEVBOX_FEATURE_FLAKES=1 devbox add ripgrep
Reinitialized existing Git repository in /Users/savil/code/jetpack/devbox/examples/testdata/go/go-1.19/.devbox/gen/flake/.git/
Installing nix packages. This may take a while...
Error: apply Nix derivation: running command /nix/var/nix/profiles/default/bin/nix profile install --profile /Users/savil/code/jetpack/devbox/examples/testdata/go/go-1.19/.devbox/nix/profile/default --extra-experimental-features nix-command flakes /Users/savil/code/jetpack/devbox/examples/testdata/go/go-1.19/.devbox/gen/flake --no-write-lock-file: exit status 1 with command stderr: warning: Git tree '/Users/savil/code/jetpack/devbox/examples/testdata/go/go-1.19/.devbox/gen/flake' is dirty
this derivation will be built:
  /nix/store/jp2yvrsx3iy7izmp297yh56nz9knw2y6-devbox-shell-env.drv
these 2 paths will be fetched (1.65 MiB download, 6.77 MiB unpacked):
  /nix/store/hisnvr8mcci3si2fsjs6r7s3h0r9yh5i-ripgrep-13.0.0
  /nix/store/qs2pncr051xkd9kq9l8mjalj7dqww6k8-pcre2-10.40
copying path '/nix/store/qs2pncr051xkd9kq9l8mjalj7dqww6k8-pcre2-10.40' from 'https://cache.nixos.org'...
copying path '/nix/store/hisnvr8mcci3si2fsjs6r7s3h0r9yh5i-ripgrep-13.0.0' from 'https://cache.nixos.org'...
building '/nix/store/jp2yvrsx3iy7izmp297yh56nz9knw2y6-devbox-shell-env.drv'...
error: files '/nix/store/xi7frhzjyxi7862wj54pxsw55xyzdg8y-devbox-shell-env/bin/go' and '/nix/store/za0pnll14dv61b124n6xxnjapd150lcr-go-1.19.4/share/go/bin/go' have the same priority 5; use 'nix-env --set-flag priority NUMBER INSTALLED_PKGNAME' or type 'nix profile install --help' if using 'nix profile' to find out howto change the priority of one of the conflicting packages (0 being the highest priority)

```

Reply:
I'm also a little confused about this. What's the difference between a devbox lockfile and the flake lockfile? Is it just to hide the flakes implementation?

Reply:
there's no difference. We copy flake.lock to devbox.lock, because the flake lockfile is generated in an inner git repository. We use an inner git repository to workaround the concern in this [issue](https://github.com/NixOS/nix/issues/3121).